src/tm_devices/drivers/pi/pi_device.py

```diff
@@ -816,7 +818,12 @@ def _cleanup(self) -> None:
 
     def _close(self) -> None:
         """Close this device and all its used resources and components."""
-        self._visa_resource.close()
+        try:
+            self._visa_resource.close()
+        except VisaIOError as error:
+            warnings.warn(
+                f"Error encountered while closing the visa resource: \n {error}", stacklevel=2
```

Comment:
Why the extra whitespace before and after the newline?

Reply:
```suggestion
                f"Error encountered while closing the visa resource:\n{error}", stacklevel=2
```

Reply:
Good catch, no particular reason